default-build-spec/main.yml

```diff
@@ -0,0 +1,11 @@
+---
+version: 1
+docker:
+  repositoryPrefix: djmitche/taskcluster-
+
+# Note: All services must have all lowercase names!
+services:
+  - name: ping
+    # note: this is a relative reference, which isn't a great idea as it means the
+    # build may include a different version in different runs
+    source: https://github.com/djmitche/taskcluster-ping#master
```

Comment:
I would suggest you design all of the input here, before you try to implement anything.

Maybe combined with a pseudo code implementation written in bullet points or something. Just so that we have clue what it'll look like..